crypto/s2n_openssl.h

```diff
@@ -51,6 +51,8 @@
 #endif
 
 bool s2n_libcrypto_is_openssl(void);
+bool s2n_libcrypto_is_openssl_fips(void);
 bool s2n_libcrypto_is_awslc();
+bool s2n_libcrypto_is_awslc_fips(void);
```

Comment:
These methods are just for convenience, but since we generally talk about the fips version of libcryptos like they're separate libcryptos, I think they're helpful and simplify common conditionals.